src/Features/Core/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -244,12 +244,16 @@ private void ProcessEvents(WorkspaceChangeEventArgs args, IAsyncToken asyncToken
 
             private void OnDocumentOpened(object sender, DocumentEventArgs e)
             {
-                EnqueueWorkItem(e.Document, InvocationReasons.DocumentOpened);
+                var asyncToken = _listener.BeginAsyncOperation("OnDocumentOpened");
```

Comment:
this is a bit unrelated to esent but found while working on this. I missed to put this in event queue so work ends up happened on UI thread where workspace event is happening.